app/jobs/invoices/abacus/transmit_all_members_job.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+#
+class Invoices::Abacus::TransmitAllMembersJob < BaseJob
+  BATCH_SIZE = 1000
+  SLICE_SIZE = 50
+  PARALLEL_THREADS = 4
+  RELEVANT_ATTRIBUTES = Invoices::Abacus::Subject::RELEVANT_ATTRIBUTES # TODO NEEDS CLARIFICATION %i[id first_name last_name email abacus_subject_key]
+  self.max_run_time = 12.hours
+
+  def perform
+    process_members
+  end
+
+  def error(_job, exception, payload = parameters)
+    HitobitoLogEntry.create!(
+      level: :error,
+      category: "rechnungen",
+      message: exception.message,
+      subject: nil,
+      payload: payload
+    )
+    super
+  end
+
+  def failure(job)
+    create_failure_log_entry(job.last_error)
+  end
+
+  private
+
+  def process_members
+    member_ids.in_batches(of: BATCH_SIZE) do |batch|
+      slices = batch.pluck(:id).each_slice(SLICE_SIZE).to_a
+      Parallel.map(slices, in_threads: PARALLEL_THREADS) do |slice|
+        raise Parallel::Break, slice if check_terminated
+        ActiveRecord::Base.connection_pool.with_connection do
+          transmit_slice(slice)
+        end
+      end
+    end
+  end
+
+  def member_ids
+    Person.joins(:roles)
+      .where(roles: {type: [Group::SektionsMitglieder::Mitglied.sti_name, Group::SektionsNeuanmeldungenNv::Neuanmeldung.sti_name]})
+      .where.not(data_quality: :error)
+      .order(:id)
+      .select(:id)
+  end
+
+  def transmit_slice(slice)
+    people = Person.where(id: slice).select(*RELEVANT_ATTRIBUTES)
```

Comment:
Hier oder in der Konstante in dieser Klasse müssten dann id und abacus_subject_key zusätzlich mitgegeben werden.